### DISCUSSION TITLE
Sketching shapes using deck.gl

### DISCUSSION THREAD

**Discussion by sandeep0391** [visgl/deck.gl#6749] [Q&A]
Hi, Is it possible to sketch/draw different shapes (rectangle, polygon, points etc.) using mouse/touchpad with the help of deck.gl ? Like shown in below screenshot - 

![image](https://user-images.githubusercontent.com/35479612/157458149-d8b4e7d8-f910-4ffd-bfd9-02d8597e6a9a.png)

**Comment by Pessimistress** [comment]
[nebula.gl](https://nebula.gl)

**Comment by sandeep0391** [comment]
Thank you for the response.
Correct me if otherwise nebula.gl is to edit a already existing structure and not to draw from scratch.

if otherwise can you please provide more details.

**Reply by ibgreen** [reply]
Did you actually look at the nebula.gl website? https://nebula.gl/geojson-editor/ 

- use the select and modify features...

**Reply by sandeep0391** [reply]
I am able to draw using EditGeoJsonLayer but how can we achieve modify feature using selection Layer. I have tried by adding selection layer to the existing layers but its not working.

for selection layer I am referring to link - https://nebula.gl/docs/api-reference/layers/selection-layer
 and below is how EditGeoJsonLayer and selection layer looks like -

![image](https://user-images.githubusercontent.com/35479612/159421592-b62ef400-1332-4b00-82f5-1720b4c45027.png)

I have tried by making EditGeoJsonLayer and selection layer as pickable: true.

can you please guide here if I am missing something.

Thanks

**Comment by youwch** [comment]
But  it cannot stop drawing polygon in nebula.gl, see https://github.com/uber/nebula.gl/issues/744.
@Pessimistress
